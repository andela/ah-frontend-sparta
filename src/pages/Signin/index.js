import React from 'react';
import SigninForm from '../../components/SigninForm';
import { FaceBook, Google, Twitter } from '../../assets/icons';

import logo from '../../assets/images/logo.png';
import '../../assets/scss/Login.scss';

class LoginPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      signinStateButton: "active",
      signupStateButton: "inactive"
    };
  }

  onSigninToggle() {
    this.setState({
      signinStateButton: "active",
      signupStateButton: "inactive"
    });
  }

  onSignupToggle() {
    this.setState({
      signinStateButton: "inactive",
      signupStateButton: "active"
    });
  }

  render() {
    return (
      <div className="outer">
        <div className="innerHalf1">
          <img src={logo} alt="Logo" className="logo" />
          <div className="intro">Welcome Back.</div>
        </div>
        <div className="innerHalf2">
          <div>
            <div className="innerHalfContent">
              <div className="socialDiv">
                Continue with <FaceBook /> <Twitter /> <Google />{" "}
              </div>
              <div className="signStr">Or</div>
              <SigninForm />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default LoginPage;
