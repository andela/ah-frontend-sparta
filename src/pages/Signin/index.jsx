import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import SigninForm from '../../components/SigninForm';
import { FaceBook, Google, Twitter } from '../../assets/icons';
import { signInUser } from '../../actions/signupSigninActions';
import logo from '../../assets/images/logo.png';
import '../../assets/scss/Login.scss';

export class Signin extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { loginUser, history } = this.props;
    const { email, password } = this.state;
    const data = {
      user: {
        email,
        password,
      },
    };
    loginUser(data, history);
  };

  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  render() {
    const { email, password } = this.state;
    return (
      <div className="outer">
        <div className="innerHalf1">
          <img src={logo} alt="Logo" className="logo" />
          <div className="intro">Welcome Back.</div>
        </div>
        <div className="innerHalf2">
          <div>
            <div className="innerHalfContent">
              <div className="socialDiv">
                Continue with
                <FaceBook />
                {' '}
                <Twitter />
                {' '}
                <Google />
              </div>
              <div className="signStr">Or</div>
              <SigninForm
                onChange={this.onChange}
                handleSubmit={this.handleSubmit}
                email={email}
                password={password}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Signin.propTypes = {
  loginUser: PropTypes.func,
  history: PropTypes.func,
};

Signin.defaultProps = {
  loginUser: () => {},
  history: () => {},
};

export default withRouter(
  connect(
    null,
    {
      loginUser: signInUser,
    },
  )(Signin),
);
