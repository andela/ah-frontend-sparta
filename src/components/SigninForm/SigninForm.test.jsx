import React from 'react';
import renderer from 'react-test-renderer';
import Signin from './index';

jest.mock('react-router-dom');
test('snapshot test', () => {
  const wrapper = renderer.create(<Signin />);
  expect(wrapper).toMatchSnapshot();
});
