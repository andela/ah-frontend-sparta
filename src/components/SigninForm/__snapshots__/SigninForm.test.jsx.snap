// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Array [
  <div
    className="signStr spacebottom"
  >
    Login to your account
  </div>,
  <form
    onSubmit={[Function]}
  >
    <div
      className="inputContain"
    >
      <input
        className="input"
        name="email"
        onChange={[Function]}
        placeholder="EMAIL"
        type="email"
        value=""
      />
    </div>
    <div
      className="inputContain"
    >
      <input
        className="input"
        name="password"
        onChange={[Function]}
        placeholder="PASSWORD"
        type="password"
        value=""
      />
    </div>
    <div
      className="forgot"
    >
      <span
        data-target="#forgotPassword"
        data-toggle="modal"
      >
        Forgot password?
      </span>
      <div>
        <div
          aria-hidden="true"
          aria-labelledby="exampleModalLabel"
          className="modal fade"
          id="forgotPassword"
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog custom-dialog"
            role="document"
          >
            <div
              className="modal-content customise_model"
            >
              <div
                className="modal-header"
              >
                <h3
                  className="modal-title"
                  id="exampleModalLabel"
                >
                  Authors Haven
                </h3>
                <button
                  aria-label="Close"
                  className="close"
                  data-dismiss="modal"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                  >
                    ×
                  </span>
                </button>
              </div>
              <div
                className="modal-body"
              >
                <div
                  className="password_reset"
                >
                  <div
                    className="form-group"
                  >
                    <label
                      className="col-form-label form_label"
                      htmlFor="email"
                      id="hfjfh"
                    >
                      <h5>
                        Forgotten Password?
                      </h5>
                      <h6
                        className="paragraph-infor"
                      >
                        Please enter your Email address. A link with instructions will be sent to you to reset your Password
                      </h6>
                      <input
                        className="form-control"
                        id="recipient-email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter your email"
                        required={true}
                        type="email"
                        value=""
                      />
                    </label>
                  </div>
                  <div
                    className="modal-footer"
                  >
                    <button
                      className="btn btn-primary send-button"
                      data-dismiss="modal"
                      onClick={[Function]}
                      type="button"
                    >
                      Click to send link to email
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Signin
    </button>
    <div
      className="other"
    >
      <span>
        Don’t have an account?
         
      </span>
    </div>
  </form>,
]
`;
